docs/fundamentals/orleans.md

```diff
@@ -0,0 +1,192 @@
+
+To participate in an Orleans cluster, reference the Orleans resource from your service project, either using `WithReference(orleans)` to participate as an Orleans server, or `WithReference(orleans.AsClient())` to participate as a client.
+When you reference the Orleans resource from your service, those resources will also be referenced:
+
+```csharp
+// Add the Orleans resource to the "silo" project:
+// As described above, this implicitly add references to the required resources.
+// In this case, that is the 'clusteringTable' resource declared earlier.
+builder.AddProject<Projects.OrleansServer>("silo")
+       .WithReference(orleans);
+```
+
+Putting that all together, here is an example of an Aspire AppHost project which includes an Orleans resource with clustering and storage, and an Orleans server and client project resource.
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+// Add the resources which we will use for Orleans clustering and grain state storage.
+var storage = builder.AddAzureStorage("storage").RunAsEmulator();
+var clusteringTable = storage.AddTables("clustering");
+var grainStorage = storage.AddBlobs("grainstate");
+
+// Add the Orleans resource to the Aspire DistributedApplication builder, then configure it with Azure Table Storage for clustering
+// and Azure Blob Storage for grain storage.
+var orleans = builder.AddOrleans("my-app")
+                     .WithClustering(clusteringTable)
+                     .WithGrainStorage("Default", grainStorage);
+
+// Reference the Orleans resource from the 'silo' project so that it can join the Orleans cluster as a service.
+builder.AddProject<Projects.OrleansServer>("silo")
+       .WithReference(orleans)
+       .WithReplicas(3);
+
+// Reference the Orleans resource as a client from the 'frontend' project so that it can connect to the Orleans cluster.
+builder.AddProject<Projects.OrleansClient>("frontend")
+       .WithReference(orleans.AsClient())
+       .WithExternalHttpEndpoints()
+       .WithReplicas(3);
+
+// Build and run the application.
+using var app = builder.Build();
+await app.RunAsync();
+```
+
+To consume the Aspire Orleans resource from an Orleans server project, there are a few steps:
+
+1. Add the relevant Aspire components. In our example, we are using _Aspire.Azure.Data.Tables_ and _Aspire.Azure.Storage.Blobs_.
+2. Add the Orleans provider packages for those Aspire components. In our example, we are using _Microsoft.Orleans.Persistence.AzureStorage_ and _Microsoft.Orleans.Clustering.AzureStorage_
+3. Add Orleans to the host application builder.
+
+In OrleansServer's _Program.cs_ file, we must configure the Aspire components we are using and add Orleans to the host builder. Note that the names provided must match the names used in the Aspire AppHost project: "clustering" for the clustering provider, and "grainstate" for the grain state storage provider:
```

Comment:
Where did this project come from? While the article has a lot of great detail, we're just jumping into a project that we didn't mention was created. 😬